include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -681,13 +643,22 @@ __radix_sort_reorder_submit(_ExecutionPolicy&& __exec, ::std::size_t __segments,
                 // 1. create a private array for storing offset values
                 //    and add total offset and offset for compute unit for a certain radix state
                 _OffsetT __offset_arr[__radix_states];
```

Comment:
What do you think about storing global offsets in local memory, filling and scanning it collaboratively? Since we are using a single sub-group as a work-group, it should not require synchronization.

Reply:
I suggest considering the next steps of improvements as the next change and the next PR, out of scope of the current PR.

Reply:
I agree with you.

It is not obvious if that change will make any performance improvement. Moreover, we must ensure synchronization according to sycl spec:

> To maximize portability across devices, developers should not assume that work-items within a sub-group execute in any particular order, that work-groups are subdivided into sub-groups in a specific way, nor that two sub-groups within a work-group will make forward progress with respect to one another.